File: TOI-Zero_68/A2/021/021.cpp
#include<bits/stdc++.h>
using namespace std;

#define ll long long

int main(){
	ios::sync_with_stdio(0);
	cin.tie(0);
	int n,k;
	cin >> n >> k;
    ll in1[n],out1[n],in2[n],out2[n];

    for(int i=0;i<n;i++){
    	cin >> in1[i];
	}
	for(int i=0;i<n;i++){
    	cin >> in2[i];
	}
	for(int i=0;i<n;i++){
    	cin >> out1[i];
	}
	for(int i=0;i<n;i++){
    	cin >> out2[i];
	}
    sort(in1,in1+n);
    sort(in2,in2+n);
    sort(out1,out1+n);
    sort(out2,out2+n);
    ll mn=LLONG_MAX;
    for(int i=0;i<=min(n,k);i++){
        ll j=k-i,mx=0;
        if(j>n){
        	continue;
		}
		for(int x=0;x<i;x++){
            mx=max(mx,in1[x]+out1[i-x-1]);
        }
        for(int x=0;x<j;x++) {
            mx=max(mx,in2[x]+out2[j-x-1]);
        }
        mn=min(mn,mx);
    }
    cout << mn;
}

File: TOI-Zero_68/A2/021/021.py
import sys

def main():
    n, k = map(int, sys.stdin.readline().split())
    
    in1 = list(map(int, sys.stdin.readline().split()))
    in2 = list(map(int, sys.stdin.readline().split()))
    out1 = list(map(int, sys.stdin.readline().split()))
    out2 = list(map(int, sys.stdin.readline().split()))
    
    in1.sort()
    in2.sort()
    out1.sort()
    out2.sort()
    
    mn = float('inf')
    
    for i in range(min(n, k) + 1):
        j = k - i
        if j > n:
            continue
            
        mx = 0
        for x in range(i):
            mx = max(mx, in1[x] + out1[i - x - 1])
        
        for x in range(j):
            mx = max(mx, in2[x] + out2[j - x - 1])
        
        mn = min(mn, mx)
    
    print(mn)

if __name__ == "__main__":
    main() 
